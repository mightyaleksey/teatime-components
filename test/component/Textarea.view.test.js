'use strict';

const React = require('react');
const Textarea = require('../../component/Textarea');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(Textarea.defaultProps.size).toBe('m');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <Textarea
      name='control'/>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <Textarea
      autoComplete='off'
      autoFocus
      className='outer'
      cols='5'
      disabled
      id='controls-id'
      maxLength='25'
      name='control'
      placeholder='it is a placeholder'
      readOnly
      rows='10'
      value='Any Code...'/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
