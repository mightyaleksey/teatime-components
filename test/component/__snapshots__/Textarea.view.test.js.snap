// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<textarea
  autoComplete={undefined}
  autoFocus={undefined}
  className="control"
  cols={undefined}
  defaultValue={undefined}
  disabled={undefined}
  id={undefined}
  maxLength={undefined}
  name="control"
  onChange={[Function]}
  placeholder={undefined}
  readOnly={undefined}
  rows="3"
  value={undefined}
/>
`;

exports[`Renders correctly with defined props 1`] = `
<textarea
  autoComplete="off"
  autoFocus={true}
  className="control outer"
  cols="5"
  defaultValue={undefined}
  disabled={true}
  id="controls-id"
  maxLength="25"
  name="control"
  onChange={[Function]}
  placeholder="it is a placeholder"
  readOnly={true}
  rows="10"
  value="Any Code..."
/>
`;
